Interface based @EachProperty with validation doesn't work

If you use `@EachPropery` on an interface and apply validation rules the configuration fails to load this is because validation is applied before the bean is registered with the context as a singleton

### Steps to Reproduce

Define:

```java

import io.micronaut.context.annotation.EachProperty;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;

@EachProperty("owners")
public interface OwnerConfiguration {
    @NotBlank
    String getName();
    @Min(18)
    int getAge();
}
```

And add configuration properties:

```
    public Map<String, String> getProperties() {
        return Map.of(
                "owners.fred.name", "Fred",
                "owners.fred.age", "35",
                "owners.barney.name", "Barney",
                "owners.barney.age", "30"
        );
    }
```

### Expected Behaviour

Should work

### Actual Behaviour

Fails with:

```
Caused by: io.micronaut.core.value.PropertyNotFoundException: No property found for name [owners.*.name] and type [java.lang.String]
	at io.micronaut.runtime.context.env.ConfigurationIntroductionAdvice.lambda$intercept$1(ConfigurationIntroductionAdvice.java:109)
	at java.base/java.util.Optional.orElseThrow(Optional.java:408)
	at 
```

### Environment Information


- **Micronaut Version:** 2.2.0

--- a/inject-java/src/test/groovy/io/micronaut/inject/configproperties/itfce/MyConfig.java
+++ b/inject-java/src/test/groovy/io/micronaut/inject/configproperties/itfce/MyConfig.java
@@ -1,0 +1,13 @@
+package io.micronaut.inject.configproperties.itfce;
+
+import io.micronaut.context.annotation.ConfigurationProperties;
+import io.micronaut.context.annotation.Requires;
+
+import javax.validation.constraints.NotBlank;
+
+@ConfigurationProperties("my.config")
+@Requires(property = "my.config")
+public interface MyConfig {
+    @NotBlank
+    String getName();
+}

--- a/inject-java/src/test/groovy/io/micronaut/inject/configproperties/itfce/MyEachConfig.java
+++ b/inject-java/src/test/groovy/io/micronaut/inject/configproperties/itfce/MyEachConfig.java
@@ -1,0 +1,13 @@
+package io.micronaut.inject.configproperties.itfce;
+
+import io.micronaut.context.annotation.EachProperty;
+import io.micronaut.context.annotation.Requires;
+
+import javax.validation.constraints.NotBlank;
+
+@EachProperty(value = "my.config", primary = "default")
+@Requires(property = "my.config")
+public interface MyEachConfig {
+    @NotBlank
+    String getName();
+}

--- a/inject-java/src/test/groovy/io/micronaut/inject/configproperties/itfce/ValidatedInterfaceConfigPropsSpec.groovy
+++ b/inject-java/src/test/groovy/io/micronaut/inject/configproperties/itfce/ValidatedInterfaceConfigPropsSpec.groovy
@@ -1,0 +1,43 @@
+package io.micronaut.inject.configproperties.itfce
+
+import io.micronaut.context.ApplicationContext
+import io.micronaut.context.exceptions.BeanInstantiationException
+import io.micronaut.inject.qualifiers.Qualifiers
+import spock.lang.Specification
+
+class ValidatedInterfaceConfigPropsSpec extends Specification {
+
+    void 'test validated interface config with invalid config'() {
+        given:
+        ApplicationContext context = ApplicationContext.run(
+                'my.config.name':'',
+                'my.config.foo.name':'',
+                'my.config.default.name':'',
+                'my.config.foo.nested.bar.name':'',
+        )
+
+        when:
+        context.getBean(MyConfig)
+
+        then:
+        def e = thrown(BeanInstantiationException)
+        e.message.contains('MyConfig.getName - must not be blank')
+
+        when:
+        context.getBean(MyEachConfig, Qualifiers.byName("foo"))
+
+        then:
+        e = thrown(BeanInstantiationException)
+        e.message.contains('MyEachConfig.getName - must not be blank')
+
+        when:
+        context.getBean(MyEachConfig)
+
+        then:
+        e = thrown(BeanInstantiationException)
+        e.message.contains('MyEachConfig.getName - must not be blank')
+
+        cleanup:
+        context.close()
+    }
+}

--- a/runtime/src/main/java/io/micronaut/runtime/context/env/ConfigurationIntroductionAdvice.java
+++ b/runtime/src/main/java/io/micronaut/runtime/context/env/ConfigurationIntroductionAdvice.java
@@ -18,42 +18,47 @@ package io.micronaut.runtime.context.env;
 import io.micronaut.aop.MethodInterceptor;
 import io.micronaut.aop.MethodInvocationContext;
 import io.micronaut.context.BeanContext;
-import io.micronaut.context.BeanRegistration;
+import io.micronaut.context.Qualifier;
 import io.micronaut.context.annotation.Property;
+import io.micronaut.context.annotation.Prototype;
 import io.micronaut.context.env.Environment;
 import io.micronaut.core.annotation.Internal;
 import io.micronaut.core.bind.annotation.Bindable;
+import io.micronaut.core.naming.Named;
 import io.micronaut.core.type.Argument;
 import io.micronaut.core.type.ReturnType;
 import io.micronaut.core.value.PropertyNotFoundException;
 
-import javax.inject.Singleton;
 import java.util.Optional;
 
 /**
  * Internal introduction advice used to allow {@link io.micronaut.context.annotation.ConfigurationProperties} on interfaces. Considered internal and not for direct use.
  *
  * @author graemerocher
- * @since 1.3.0
  * @see ConfigurationAdvice
  * @see io.micronaut.context.annotation.ConfigurationProperties
+ * @since 1.3.0
  */
-@Singleton
+@Prototype
 @Internal
 public class ConfigurationIntroductionAdvice implements MethodInterceptor<Object, Object> {
     private static final String MEMBER_BEAN = "bean";
     private static final String MEMBER_NAME = "name";
     private final Environment environment;
     private final BeanContext beanContext;
+    private final String name;
 
     /**
      * Default constructor.
+     *
+     * @param qualifier   The qualifier
      * @param environment The environment
-     * @param beanContext  The bean locator
+     * @param beanContext The bean locator
      */
-    ConfigurationIntroductionAdvice(Environment environment, BeanContext beanContext) {
+    ConfigurationIntroductionAdvice(Qualifier<?> qualifier, Environment environment, BeanContext beanContext) {
         this.environment = environment;
         this.beanContext = beanContext;
+        this.name = qualifier instanceof Named ? ((Named) qualifier).getName() : null;
     }
 
     @Override
@@ -81,9 +86,7 @@ public class ConfigurationIntroductionAdvice implements MethodInterceptor<Object
             }
             boolean iterable = property.indexOf('*') > -1;
             if (iterable) {
-                final BeanRegistration<Object> registration = beanContext.findBeanRegistration(context.getTarget()).orElse(null);
-                if (registration != null) {
-                    final String name = registration.getIdentifier().getName();
+                if (name != null) {
                     property = property.replace("*", name);
                 }
             }

